teensytoany/teensytoany.py

```diff
@@ -276,42 +277,58 @@ def i2c_write_read(self,
-            self._ask(
-                f"i2c_write_no_register_uint8 0x{address:02x} 0x{data:02x}")
-        elif len(payload) == 3:
-            # uint8
-            register_address = int.from_bytes(
-                payload[0:2], byteorder='big', signed=False)
-            data = int.from_bytes(
-                payload[2:3], byteorder='big', signed=False)
-            self._ask(
-                f"i2c_write_uint8 "
-                f"0x{address:02x} 0x{register_address:04x} 0x{data:02x}")
-        elif len(payload) == 4:
-            register_address = int.from_bytes(
-                payload[0:2], byteorder='big', signed=False)
-            data = int.from_bytes(
-                payload[2:4], byteorder='big', signed=False)
-            self._ask(
-                f"i2c_write_uint16 "
-                f"0x{address:02x} 0x{register_address:04x} 0x{data:04x}")
-        else:
-            raise NotImplementedError()
+    def i2c_write_payload(self, address: int, register_address: int, payload: Sequence) -> None:
 
-    def i2c_read_bytes(self, address: int, num_bytes: int=1) -> Sequence:
-        if num_bytes != 1:
-            raise NotImplementedError()
-        returned = self._ask(f"i2c_read_no_register_uint8 0x{address:02x}")
-        register_data = int(returned, base=0)
-        return int.to_bytes(
-            int(register_data),
-            length=num_bytes, byteorder='big',
-            signed=False)
+        if self.version >= Version("0.0.13"):
+            data =  ' '.join([f"0x{val:02x}" for val in payload])
+            cmd = f"i2c_write_payload 0x{address:02x} 0x{register_address:02x} {data}"
+            self._ask(cmd)
+
+        else:
+            if len(payload) == 1:
+                # Trying to write the network chips
+                data = int(payload[0])
+                self._ask(
+                    f"i2c_write_no_register_uint8 0x{address:02x} 0x{data:02x}")
+            elif len(payload) == 3:
+                # uint8
+                register_address = int.from_bytes(
+                    payload[0:2], byteorder='big', signed=False)
+                data = int.from_bytes(
+                    payload[2:3], byteorder='big', signed=False)
+                self._ask(
+                    f"i2c_write_uint8 "
+                    f"0x{address:02x} 0x{register_address:04x} 0x{data:02x}")
+            elif len(payload) == 4:
+                register_address = int.from_bytes(
+                    payload[0:2], byteorder='big', signed=False)
+                data = int.from_bytes(
+                    payload[2:4], byteorder='big', signed=False)
+                self._ask(
+                    f"i2c_write_uint16 "
+                    f"0x{address:02x} 0x{register_address:04x} 0x{data:04x}")
+            else:
+                raise NotImplementedError()
+        
+
+    def i2c_read_payload(self, address: int, register_address: int, num_bytes: int) -> Sequence:
+
+        if self.version >= Version("0.0.13"):
```

Comment:
```suggestion
        if self.version > Version("0.0.13"):
```

Doesn't this only work for 0.0.14? 0.0.13+1230912830 (when you are developing) might be accepted by the `>`. I think. In either case, that isn't super critical. It should for sure give the correct results for released version:
* 0.0.13 and below
* 0.0.14 (newly released)